src/lib/pipe_lib/broadcast_pipe.ml

```diff
@@ -0,0 +1,134 @@
+open Core_kernel
+open Async_kernel
+
+type 'a t =
+  { mvar: 'a option Mvar.Read_only.t
```

Comment:
Why is mvar paraterized by an 'a option? Mvar is essentially an option itself. So we basically dealing with  'a option option? This doesn't seem right.

Reply:
I think this is what we want. Several listeners (that aren’t written yet: the transaction pool and snark pool) need to receive events when the transition frontier goes away. Additionally, taking from the mvar is done immediately by the broadcast pipe so the mvar is always empty essentially.